### PR TITLE
GH-91 dcss reconnect

### PR DESCRIPTION
**Pull request by dsclassen**
A simple change to `tcpip.py` allows dcss to reconnect if it gets restarted while DHS is running.

### CODE REVIEW

**Comment by dsclassen** [comment]
I guess I created this branch from gh96_prepare_for_packaging instead of master...? hmmm?

File: pydhsfw/tcpip.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ def _disconnect(self):
                 self._stream_reader.socket = None
                 self._stream_writer.socket = None
                 if sock:
-                    sock.shutdown(socket.SHUT_RDWR)
+                    #sock.shutdown(socket.SHUT_RDWR)
                     sock.close()
```
Comment (gcmullen) [inline pydhsfw/tcpip.py:251]: https://stackoverflow.com/questions/409783/socket-shutdown-vs-socket-close

There are scenarios where `close()` will not work due to reference counting. In this scenario just a `close()` appears to be working, but other cases may not work. Generally we really do want to shutdown the socket nicely so the other end has an opportunity to gracefully disconnect and be available to us sooner for reconnecting. Our local socket reference counting will eventually sort itself out through garbage collection. Better to test the socket status prior to calling `shutdown()` or wrap it in a try/except explicitly catching `OSError: [Errno 57] Socket is not connected`.

**Review by gcmullen — CHANGES_REQUESTED** [review]

File: pydhsfw/tcpip.py (hunk 1 above)
Comment (dsclassen) [reply pydhsfw/tcpip.py]: Not sure how to test socket status so I added a try/except/finally block.

File: pydhsfw/tcpip.py [hunk 2]
```diff
@@ -247,8 +247,12 @@ def _disconnect(self):
                 self._stream_reader.socket = None
                 self._stream_writer.socket = None
                 if sock:
-                    sock.shutdown(socket.SHUT_RDWR)
-                    sock.close()
+                    try:
+                        sock.shutdown(socket.SHUT_RDWR)
+                    except:
```
Comment (gcmullen) [inline pydhsfw/tcpip.py:252]: I was thinking of being more specific on the `except OSError`, but that should work.

**Review by gcmullen — APPROVED** [review]
Good.

File: pydhsfw/tcpip.py (hunk 2 above)
Comment (dsclassen) [reply pydhsfw/tcpip.py]: OK. I think this is done. I'd like to merge.

**Pull request merged**